### PR TITLE
feat: refactor the interactive demo section on the landing page + misc. styling updates

### PR DESCRIPTION
**Pull request by mindapivessa**
**What changed? Why?**
- Added `ComponentPreview` and `LandingFooter` components
- `ComponentPreview` enables users to toggle between different components. For each component, they can see interactive preview and ready-to-copy example code snippet.
- Updated the `Tweets` to horizontally stack the cards. Will follow up animated horizontal scroll in another PR.
- Updated background colors to increase polish.

**Acceptance criteria**
- [x] Each live demo works end-to-end (apart from Swap)
- [x] Code snippets render properly and accurately 
- [x] Copying works
- [ ] Mobile responsive
- [ ] Each link in the Footer links out to the right destination

**Notes to reviewers**
- git clone https://github.com/mindapivessa/onchainkit.git
- Switch to `mind/update-landing-page-1022` branch to pull the changes
- Try running it locally please! My vercel build is still failing :( 
- Please review the accuracy code snippets! The users should be able to copy and use it minimal modification.

**How has it been tested?**

Desktop

https://github.com/user-attachments/assets/e079a0e6-d305-499a-bec2-1a23d1ede6aa

Mobile/responsive

https://github.com/user-attachments/assets/c77c2fe5-6877-4717-ae69-86e7297cfdb5

### CODE REVIEW

**Review by 0xAlec — APPROVED** [review]
approving this to authorize the vercel deploy for now

File: site/docs/components/landing/ComponentPreview.tsx [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+  activeTab,
+  activeSubTab,
+  setActiveSubTab,
+  copiedIndex,
+  copyToClipboard,
+}: PreviewContainerProps) {
+  const ActiveComponent = components[activeTab].component;
+
+  return (
+    <div className="w-full rounded-lg border-[1px] border-zinc-200 lg:w-[800px] lg:flex-shrink-0 dark:border-zinc-900">
```
Comment (dschlabach) [inline site/docs/components/landing/ComponentPreview.tsx:123]: Why do we need to use an arbitrary value here? `border-px` is built into Tailwind so should work out of the box.

I did try locally though and it didn't work, not sure what's going on with the Tailwind setup

File: site/docs/components/landing/ComponentPreview.tsx [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+      <div className="relative h-[400px] w-full overflow-hidden rounded-lg bg-zinc-100 lg:h-[600px] dark:bg-[#0F0F0F]">
+        <div className="flex items-center justify-between border-zinc-200 border-b dark:border-zinc-900">
+          <div className="flex">
+            <button
+              type="button"
+              className={`mt-2 ml-2 px-4 py-2 text-sm ${
+                activeSubTab === 'preview'
+                  ? 'border-indigo-600 border-b-2 bg-zinc-100 text-zinc-950 dark:border-indigo-400 dark:bg-[#0F0F0F] dark:text-zinc-50'
```
Comment (dschlabach) [inline site/docs/components/landing/ComponentPreview.tsx:131]: Same here, we should avoid magic numbers/variables in the codebase (recommended way of doing this is to define in the tailwind config file

**Comment by cpcramer** [comment]
The top left logo "click" surface area is much larger than expected

Screenshot 2024-10-23 at 9 10 49 AM

**Comment by cpcramer** [comment]
The Fund button seems a little small?

Screenshot 2024-10-23 at 9 12 32 AM

File: site/docs/components/landing/ComponentPreview.tsx (hunk 1 above)
Comment (mindapivessa) [reply site/docs/components/landing/ComponentPreview.tsx]: addressed this here: https://github.com/coinbase/onchainkit/pull/1482/commits/ee94371f085a6d33d8586eb3285683c74f6f6e36

File: site/docs/components/landing/CheckoutDemo.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import App from '../App.tsx';
+
+export const checkoutDemoCode = `
+  import { 
+      Checkout, 
+      CheckoutButton
+  } from '@coinbase/onchainkit/checkout';
+ 
+  <Checkout productId='my-product-id' > 
+    <CheckoutButton coinbaseBranded/>
```
[the remark this answers is not in the document]
Comment (0xAlec) [reply site/docs/components/landing/CheckoutDemo.tsx:11]: it's true if you specify the prop

[the remark this answers is not in the document]
Comment (mindapivessa) [reply site/docs/components/landing/CheckoutDemo.tsx]: addressed here https://github.com/coinbase/onchainkit/pull/1482/commits/ca8d92daaf690e4eee7b09c956bcb10b0f89fa2d

**Review by cpcramer — APPROVED** [review]

**Pull request merged**